docs/library/rp2.rst

```diff
@@ -71,6 +71,79 @@ For running PIO programs, see :class:`rp2.StateMachine`.
     This exception is raised from `asm_pio()` or `asm_pio_encode()` if there is
     an error assembling a PIO program.
 
+Each state machine has a set of instructions- which allow for a wide range of
+behaviourial or location changes to the data. They are written in assembly
+language which is later used in a MicroPython program to perform specific
+tasks. These instructions are -
+
+jmp(cond, lebel=None)
+	condition : 0-7
+	<delay_value> : 0-31
+	target : instruction offset of label within the program
+    see sec 3.4.2 of RPi docs for details
+
+wait(polarity, src, index)
+     polarity: 0-1
+     src: gpio, pin, irq
+     index: 0-31
+     wait for high/low on pin 0-31 or irq 0-31
+     see sec 3.4.3 of RPi docs for details
+
+in(src, data)
+	src : pin, X, Y, ISR, OSR(all 0-31)
```

Comment:
these all need to be lower case: x, y, isr, osr.  also it's `in_` to not clash with the `in` keyword